components/helm-broker/internal/addon/provider/client.go

```diff
@@ -0,0 +1,128 @@
+package provider
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/ghodss/yaml"
+	"github.com/kyma-project/kyma/components/helm-broker/internal"
+	"github.com/kyma-project/kyma/components/helm-broker/internal/addon"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+// Client wraps the concrete getters and provide common functionality for converting the raw bundles into models.
+type Client struct {
+	getters             map[string]RepositoryGetter
+	specifiedSchemRegex *regexp.Regexp
+	log                 *logrus.Entry
+	bundleLoader        addonLoader
+	concreteGetter      RepositoryGetter
+}
+
+// NewClient returns new instance of Client
+func NewClient(concreteGetter RepositoryGetter, bundleLoader addonLoader, log logrus.FieldLogger) (*Client, error) {
+	specifiedSchemRegex, err := regexp.Compile(`^([A-Za-z0-9]+)::(.+)$`)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{
+		specifiedSchemRegex: specifiedSchemRegex,
+		bundleLoader:        bundleLoader,
+		log:                 log.WithField("service", "RepositoryGetter:CompleteBundleProvider"),
```

Comment:
I'd like to see what kind of getter is working (another field in the logger) - git, hg, s3 etc

Reply:
good point, implemented